pkg/apis/metal3/v1alpha1/baremetalhost_types_test.go

```diff
@@ -376,24 +376,6 @@ func TestHostNeedsDeprovisioning(t *testing.T) {
 			Expected: false,
 		},
 
-		{
-			Scenario: "externally provisioned",
```

Comment:
Don't we still need this test case?

Reply:
No, because we don't call this from any states where transitioning to ExternallyProvisioned is allowed (see the second commit message that GitHub does a very good job of hiding from reviewers).
I know that's an unsatisfying answer, due to the problems identified in your other comment.